EliasHjelm/mumsig
# Feature/filter function

Feature: filter by category on startPage (dropdowns on the left)
**Test cases**

1. click on one category option, e.g. Kött
Result:

- category option is checked and has 2 icons 2 "checked" and "cancel";
- recipeCards filtered by selected option appeared in the content-section (Resultat för heading) instead of Senaste recept

*****
2. click on another category option (2 options checked), e.g. Kött & Pasta
Result:

- category options are checked both
- recipeCards filtered by these both selected options (recipe should have both options in its categories)

***** 
3. click one more category option (e.g. Kött & Past & Fisk)
Result:

- category options are checked;
- Under Resultat för appears text Ingen träff, because there are recepts in db which contains all of these checked ingredients 

*******************
4. click on checked category option
Result:

- clicked category option is unchecked, icons "checked" and "cancel" disappeared
- Under Resultat appears recipes filtered by the rest checked category option (or test case 3)

5. click on every checked category (uncheck all, i.e. clear filter)

- clicked category options are unchecked,  icons "checked" and "cancel" disappeared
- Senaste Recept appears in the content-section instead of Resultat för

************
Try more filter combinations and alert about bugs if they occur.
Happy testing!
